### PR TITLE
chore: support next.js 15

### PR DESCRIPTION
**Pull request by siddhsuresh**
Closes: #4385 https://github.com/blitz-js/blitz/issues/4385

### What are the changes and their implications?

Upgrade packages to avoid warnings thrown from latest next.js version

### CODE REVIEW

**Comment by siddhsuresh** [comment]
!preview next-js-15

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 1]
```diff
@@ -289,8 +289,10 @@ const makeProxyToPublicData = <T extends SessionContextClass>(ctxClass: T): T =>
 export async function getBlitzContext(): Promise<Ctx> {
   try {
     const {headers, cookies} = require("next/headers")
-    const reqHeader = Object.fromEntries(headers())
-    const csrfToken = cookies().get(COOKIE_CSRF_TOKEN())
+    const cookieStore = await cookies()
```
Comment (stijnvanderlaan) [inline packages/blitz-auth/src/server/auth-sessions.ts:292]: I would wrap this in `Promise.resolve` to maintain backward compatibility with Next.js 14 and earlier versions.

File: packages/blitz-auth/src/server/auth-sessions.ts [hunk 2]
```diff
@@ -289,8 +289,10 @@ const makeProxyToPublicData = <T extends SessionContextClass>(ctxClass: T): T =>
...
+    const headersStore = await headers()
```
Comment (stijnvanderlaan) [inline packages/blitz-auth/src/server/auth-sessions.ts:293]: And here offcourse

File: apps/next13/package.json [hunk 3]
```diff
@@ -22,18 +22,18 @@
     "@tanstack/react-query": "4.0.10",
     "blitz": "2.1.3",
     "flatted": "3.2.7",
-    "next": "14.2.15",
+    "next": "15.0.1",
     "prisma": "^4.5.0",
-    "react": "18.2.0",
```
Comment (gengjiawen) [inline apps/next13/package.json:27]: https://react.dev/blog/2024/12/05/react-19 official 19 released

Comment (siddhsuresh) [reply apps/next13/package.json:27]: thanks @gengjiawen. will update the PR

Comment (siddhsuresh) [reply apps/next13/package.json:27]: done

**Review by flybayer — APPROVED** [review]

**Comment by seekshiva** [comment]
What will be the ETA for this version to be pushed to npm?

**Comment by MonsterDKing** [comment]
i test create new proyect and mount in vercel and give me this error .. 

Error: Usage of `getBlitzContext` is supported only in next.js 13.0.0 and above. Please upgrade your next.js version. at /var/task/.next/server/chunks/646.js:44:28080 at Generator.next (<anonymous>) at /var/task/.next/server/chunks/646.js:44:25680 at new Promise (<anonymous>) at eO (/var/task/.next/server/chunks/646.js:44:25500) at eD (/var/task/.next/server/chunks/646.js:44:27803) at /var/task/.next/server/chunks/646.js:44:77420 at Generator.next (<anonymous>) at /var/task/.next/server/chunks/646.js:44:77305 at new Promise (<anonymous>) { digest: '3648858347' }



this is the package.json 

`{
  "name": "testing",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "blitz dev",
    "build": "blitz build",
    "start": "blitz start",
    "studio": "blitz prisma studio",
    "lint": "eslint --ignore-path .gitignore --ext .js,.ts,.tsx .",
    "test": "vitest run --passWithNoTests",
    "test:watch": "vitest",
    "prepare": "husky install"
  },
  "prisma": {
    "schema": "db/schema.prisma"
  },
  "prettier": {
    "semi": false,
    "printWidth": 100
  },
  "lint-staged": {
    "*.{js,ts,tsx}": [
      "eslint --fix"
    ]
  },
  "dependencies": {
    "@blitzjs/auth": "2.2.0",
    "@blitzjs/next": "2.2.0",
    "@blitzjs/rpc": "2.2.0",
    "@prisma/client": "5.4.2",
    "blitz": "2.2.0",
    "formik": "2.4.6",
    "next": "14.2.15",
    "prisma": "5.4.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "secure-password": "4.0.0",
    "zod": "3.23.8"
  },
  "devDependencies": {
    "@next/env": "13.4.19",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react": "13.4.0",
    "@testing-library/react-hooks": "8.0.1",
    "@types/node": "18.11.9",
    "@types/preview-email": "2.0.1",
    "@types/react": "18.0.25",
    "@typescript-eslint/eslint-plugin": "5.42.1",
    "@vitejs/plugin-react": "2.2.0",
    "ajv": "^7.0.0",
    "eslint": "8.27.0",
    "eslint-config-next": "12.3.1",
    "eslint-config-prettier": "8.5.0",
    "husky": "8.0.2",
    "jsdom": "20.0.3",
    "lint-staged": "13.0.3",
    "prettier": "^2.7.1",
    "prettier-plugin-prisma": "4.4.0",
    "pretty-quick": "3.1.3",
    "preview-email": "3.0.7",
    "typescript": "^4.8.4",
    "vite-tsconfig-paths": "3.6.0",
    "vitest": "0.25.3"
  }
}
`

**Pull request merged**